### PR TITLE
Create Scan Benchmark

### PR DESCRIPTION
**Pull request by justinborromeo**
This PR creates a Scan query benchmark to act as a baseline for future changes to the Scan query (specifically #6088 ).  The queries are fairly simple and if anyone has any suggestions for better queries to do, lmk.  Also, I would greatly appreciate if someone could sanity-check my results (below).

The ScanQuery builder was also moved from the ScanQuery class to the Druids class to match the pattern of all the other query builders.

Not sure if this matters since this PR doesn't involve any changes to the query code but here's the results of the benchmark running in IntelliJ:

image

### CODE REVIEW

File: benchmarks/src/main/java/org/apache/druid/benchmark/query/ScanBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+
+    final List<String> dimUniformFilterVals = new ArrayList<>();
+    int resultNum = (int) (100000 * 0.1);
+    int step = 100000 / resultNum;
+    for (int i = 1; i < 100001 && dimUniformFilterVals.size() < resultNum; i += step) {
+      dimUniformFilterVals.add(String.valueOf(i));
+    }
+
+    List<String> dimHyperUniqueFilterVals = new ArrayList<>();
+    resultNum = (int) (100000 * 0.1);
```
Comment (jon-wei) [inline benchmarks/src/main/java/org/apache/druid/benchmark/query/ScanBenchmark.java]: nit: suggest `numResults` instead

File: benchmarks/src/main/java/org/apache/druid/benchmark/query/ScanBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+    final int resultNum = (int) (100000 * 0.1);
+    final int step = 100000 / resultNum;
+    for (int i = 1; i < 100001 && dimUniformFilterVals.size() < resultNum; i += step) {
+      dimUniformFilterVals.add(String.valueOf(i));
+    }
+
+    final String dimName = "dimUniform";
+    return Druids.newScanQueryBuilder()
+        .filters(new SelectorDimFilter(dimName, "3", StrlenExtractionFn.instance()))
+        .intervals(intervalSpec); // TODO
```
Comment (jon-wei) [inline benchmarks/src/main/java/org/apache/druid/benchmark/query/ScanBenchmark.java]: What's the TODO here?

File: benchmarks/src/main/java/org/apache/druid/benchmark/query/ScanBenchmark.java [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+    return Druids.newScanQueryBuilder()
+                 .filters(filter)
+                 .intervals(intervalSpec);
+  }
+
+  private static Druids.ScanQueryBuilder basicC(final BenchmarkSchemaInfo basicSchema)
+  {
+    final QuerySegmentSpec intervalSpec = new MultipleIntervalSegmentSpec(Collections.singletonList(basicSchema.getDataInterval()));
+
+    final List<String> dimUniformFilterVals = new ArrayList<>();
```
Comment (jon-wei) [inline benchmarks/src/main/java/org/apache/druid/benchmark/query/ScanBenchmark.java]: dimUniformFilterVals is unused

File: benchmarks/src/main/java/org/apache/druid/benchmark/query/ScanBenchmark.java [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+        .intervals(intervalSpec); // TODO
+  }
+
+  private static Druids.ScanQueryBuilder basicD(final BenchmarkSchemaInfo basicSchema)
+  {
+    final QuerySegmentSpec intervalSpec = new MultipleIntervalSegmentSpec(
+        Collections.singletonList(basicSchema.getDataInterval())
+    );
+
+    final List<String> dimUniformFilterVals = new ArrayList<>();
```
Comment (jon-wei) [inline benchmarks/src/main/java/org/apache/druid/benchmark/query/ScanBenchmark.java]: dimUniformFilterVals is unused

File: benchmarks/src/main/java/org/apache/druid/benchmark/query/ScanBenchmark.java [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+    for (int i = 1; i < 100001 && dimUniformFilterVals.size() < resultNum; i += step) {
+      dimUniformFilterVals.add(String.valueOf(i));
+    }
+
+    final String dimName = "dimUniform";
+
+
+    return Druids.newScanQueryBuilder()
+        .filters(new BoundDimFilter(dimName, "100", "10000", true, true, true, null, null))
+        .intervals(intervalSpec); // TODO
```
Comment (jon-wei) [inline benchmarks/src/main/java/org/apache/druid/benchmark/query/ScanBenchmark.java]: What is the TODO?

File: benchmarks/src/main/java/org/apache/druid/benchmark/query/ScanBenchmark.java (hunk 2 above)
Comment (justinborromeo) [reply benchmarks/src/main/java/org/apache/druid/benchmark/query/ScanBenchmark.java]: Oops didn't commit the changes where I removed that.

**Pull request merged**